src/include/btree_inline.h

```diff
@@ -64,6 +64,21 @@ __wt_evict_page_soon_check(WT_SESSION_IMPL *session, WT_REF *ref, bool *inmem_sp
     btree = S2BT(session);
     page = ref->page;
 
+    if (!__wt_evict_page_is_soon_or_wont_need(page))
+        return (false);
+
+    /*
+     * Don't schedule pages that will require an update-restore eviction. Wait for them to be old
+     * enough to get evicted.
+     */
+    if (F_ISSET_ATOMIC_64(S2C(session), WT_CONN_PRECISE_CHECKPOINT) &&
```

Comment:
Should this also check for `WT_EVICT_CACHE_DIRTY_HARD | WT_EVICT_CACHE_UPDATES_HARD` flags like `__evict_skip_dirty_candidate()` does?

Reply:
Yep - that's a good idea. I suspect this is too aggressive as it stands, not applying the check when the hard thresholds have been met is worth testing.

It's also worth noting that as it stands, this change is specific to disaggregated storage (which enabled precise checkpoints), so it's not relevant to ASC deployments.